flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -1097,6 +1107,16 @@ public void suspend(Throwable suspensionCause) {
 		}
 	}
 
+	public void failGlobal(Throwable cause, ExecutionAttemptID failingAttempt) {
+		final Execution failedExecution = currentExecutions.get(failingAttempt);
+		if (failedExecution != null && failedExecution.getState() == ExecutionState.RUNNING) {
+			failGlobal(cause);
+		} else {
+			LOG.warn("The failing attempt {} belongs to an already not" +
```

Comment:
Should not be a warning but debug log message since we expect this to happen (without having negative side effects).